### PR TITLE
store: Add more log

### PR DESCRIPTION
**Pull request by xiekeyi98**
### What problem does this PR solve? 

Now the log is little in `newStoreWithRetry`, it is hard to debug.
We should add more information to it.

### What is changed and how it works?

Add more logs in the `newStoreWithRerty`.

### Check List 

Tests 

 - Unit test
 - Integration test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/10019)

### CODE REVIEW

**Comment by xiekeyi98** [comment]
@winkyao  PTAL, Thanks.

File: store/store.go [hunk 1]
```diff
@@ -54,12 +55,14 @@ func New(path string) (kv.Storage, error) {
 func newStoreWithRetry(path string, maxRetries int) (kv.Storage, error) {
 	storeURL, err := url.Parse(path)
 	if err != nil {
+		logutil.Logger(context.Background()).Warn("url parse failed", zap.String("path", path), zap.Error(err))
```
Comment (winkyao) [inline store/store.go]: No need to log it here, we will print the error outside, see terror.MustNil

```
	storage, err = kvstore.New(fullPath)
	terror.MustNil(err)
```

File: store/store.go [hunk 2]
```diff
@@ -54,12 +55,14 @@ func New(path string) (kv.Storage, error) {
...
 		return nil, errors.Trace(err)
 	}
 
 	name := strings.ToLower(storeURL.Scheme)
 	d, ok := stores[name]
 	if !ok {
+		logutil.Logger(context.Background()).Warn("invalid url format, storage is not registered", zap.String("storage name", name))
```
Comment (winkyao) [inline store/store.go]: ditto

File: store/store.go [hunk 3]
```diff
@@ -69,5 +72,11 @@ func newStoreWithRetry(path string, maxRetries int) (kv.Storage, error) {
 		s, err = d.Open(path)
 		return kv.IsRetryableError(err), err
 	})
+	s, err = d.Open(path)
+	if err == nil {
```
Comment (winkyao) [inline store/store.go]: Only log when err != nil

File: store/store.go (hunk 1 above)
Comment (xiekeyi98) [inline store/store.go]:
```suggestion
```

File: store/store.go (hunk 2 above)
Comment (xiekeyi98) [reply store/store.go]:
```suggestion
```

File: store/store.go [hunk 4]
```diff
@@ -69,5 +70,11 @@ func newStoreWithRetry(path string, maxRetries int) (kv.Storage, error) {
 		s, err = d.Open(path)
 		return kv.IsRetryableError(err), err
 	})
+	s, err = d.Open(path)
```
Comment (xiekeyi98) [inline store/store.go]:
```suggestion
```

File: store/store.go [hunk 5]
```diff
@@ -69,5 +70,10 @@ func newStoreWithRetry(path string, maxRetries int) (kv.Storage, error) {
 		s, err = d.Open(path)
```
Comment (bb7133) [inline store/store.go]: How about adding some store details here? For example, the `path`.

Comment (xiekeyi98) [reply store/store.go]: Done, PTAL.

**Comment by bb7133** [comment]
LGTM

**Pull request merged**